main/src/cgeo/geocaching/ui/dialog/SimpleDialog.java

```diff
@@ -0,0 +1,446 @@
+        }
+        if (this.message != null) {
+            builder.setMessage(this.message.getText(getActivity()));
+        }
+    }
+
+    /**
+     * adjusts common dialog settings for the created dialog (e.g. title and message). Call this method after calling dialog.show()!
+     */
+    public void adjustCommons(final AlertDialog dialog) {
+        if (this.title != null) {
+            this.title.applyTo(dialog.findViewById(android.R.id.title));
+        }
+        if (this.message != null) {
+            this.message.applyTo(dialog.findViewById(android.R.id.message));
+        }
+    }
+
+    /**
+     * Use this method to create and display a 'confirmation' dialog using the data defined before using this classes' setters
+     *
+     * A confirm dialog always has at least a positive action and shows at least a positive and a negative button.
+     * Provide up to three listeners to define actions for 'positive', 'negative' and 'neutral' button.
+     */
+    public void confirm(final DialogInterface.OnClickListener positiveListener, final DialogInterface.OnClickListener... otherListeners) {
+        final DialogInterface.OnClickListener[] allListeners = ArrayUtils.addFirst(otherListeners, positiveListener);
+
+        if (allListeners.length >= 2) {
+            show(allListeners);
+        } else {
+            //"confirm" always needs at least "ok" and "cancel" button
+            final DialogInterface.OnClickListener[] newListener = new DialogInterface.OnClickListener[2];
+            newListener[0] = allListeners.length == 0 ? DO_NOTHING : allListeners[0];
+            newListener[1] =  DO_NOTHING;
+            show(newListener);
+        }
+    }
+
+     /**
+      * Use this method to create and display a simple 'show message' dialog using the data defined before using this classes' setters
+      *
+      * A show dialog just shows a message, only one positive button is mandatory (but does not need to have an action)
+      * Provide up to three listeners to define actions for 'positive', 'negative' and 'neutral' button.
+      */
+    public void show(final DialogInterface.OnClickListener... listener) {
+        final AlertDialog.Builder builder = Dialogs.newBuilder(getActivity());
+        applyCommons(builder);
+        builder.setCancelable(true);
+        if (listener == null || listener.length == 0) {
+            builder.setPositiveButton(getPositiveButton(), DO_NOTHING);
+        } else {
+            builder.setPositiveButton(getPositiveButton(), listener[0]);
+            if (listener.length > 1 && listener[1] != null) {
+                builder.setNegativeButton(getNegativeButton(), listener[1]);
+            }
+            if (listener.length > 2 && listener[2] != null) {
+                builder.setNeutralButton(getNeutralButton(), listener[2]);
+            }
+        }
+        final AlertDialog dialog = builder.create();
+        dialog.setOwnerActivity((Activity) getActivity());
```

Comment:
What is the purpose of this? If we leave out "setOwningActivity" (which is only necessary for sharing autio channels ec
tc) we could  make this whole  class based on `Context` instead of `Activity`...

Reply:
I thought about this too and in fact first implemented it this way (the unnecessary cast above is an artefact of this which I forgot to remove - I can do this). 

Then I realized that at least currently there is not a single place in c:geo using Dialogs with a context  which is not also an activity, so I can really not tell if this all would work when context is not an activity.

What would be a use case for a dialog which is NOT in the context of an activity?

Reply:
I now introduced a new creation method `ofContext` which just takes a context (not an Activity). If at any time there will be a use case for it, it can be used.

Reply:
Yes, an alert dialog needs to be called from an activity. But there might be places where we have just the context object of the activity... 🤷‍♂️ Not sure, but anyway `ofContext` sounds like a good idea 👍